pkg/storegateway/series_refs.go

```diff
@@ -891,7 +929,7 @@ func (s *loadingSeriesChunkRefsSetIterator) Next() bool {
 	}
 
 	s.currentSet = nextSet
-	if s.skipChunks && cachedSeriesID.isSet() {
+	if s.strategy.isNoChunks() && cachedSeriesID.isSet() {
```

Comment:
```suggestion
	if s.strategy.isNoChunksOnEntireBlock() && cachedSeriesID.isSet() {
```

but I also think we can just get rid of the first conditional
```suggestion
	if cachedSeriesID.isSet() {
```